### DISCUSSION TITLE
Seek to causing player to play when paused

### DISCUSSION THREAD

**Discussion by CarrieEBeaton** [katspaugh/wavesurfer.js#3432] [Q&A] [answered]
Hi, when I use seekTo, the first time it gets called it plays the audio, afterwards it is okay - is this a bug or am I doing something incorrect?  Thanks

**Comment by katspaugh** [comment]
Hmm, normally it should just set the position w/o playing the audio. Maybe you're triggering playback from some event?

**Reply by CarrieEBeaton** [reply]
It also only works when I have it inside a ready event or something like this -  should I be able to call it outside of these events and have the wavesurfer update?

this.wavesurfer.once('decode', () => {
        if (this.currentPlayerTime) {
          this.wavesurfer.seekTo(this.currentPlayerTime / this.wavesurfer.getDuration());
        }
      });

**Reply by katspaugh** [reply]
That shouldn't trigger playback on its own.

Btw you can use `this.wavesurfer.setTime(this.currentPlayerTime)` instead of `seekTo`.

**Reply by CarrieEBeaton** [reply]
When I use set time it has the same issue, that's why I went to seek.  I still have to use it inside a ready event or decode to get the wavesurfer to update, is it supposed to be called inside one of those events?  If I call it outside of those events, it doesn't update the wavesurfer.

**Reply by katspaugh** [reply, marked as answer]
Yes, it has to be inside a ready, canplay or decode event.
